server/src/main/java/org/opensearch/cluster/coordination/Coordinator.java

```diff
@@ -1447,13 +1454,12 @@ protected void onFoundPeersUpdated() {
                     foundPeers.forEach(expectedVotes::addVote);
                     expectedVotes.addVote(Coordinator.this.getLocalNode());
                     final boolean foundQuorum = coordinationState.get().isElectionQuorum(expectedVotes);
-
                     if (foundQuorum) {
                         if (electionScheduler == null) {
                             startElectionScheduler();
                         }
                     } else {
-                        closePrevotingAndElectionScheduler();
```

Comment:
why do we need to change this code path? It didnt find quorum so let it close both Prevoting & Election scheduler

Reply:
If quorum is not found then we know that the current election will not succeed so we close the prevoting round. If we close the election scheduler as well then it is possible that the quorum is found immediately (say after 100 ms) which will restart the election scheduler and increment the term. If there was a cluster state computation for a previous election on another node then that will fail.
If we do not cancel the election scheduler it will keep incrementing the failed election attempt and increase the backoff time. Even if the quorum is found immediately, the next election will start only after a certain backoff time which will allow any current election to complete.

Reply:
Reverted this change.